common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -582,7 +583,18 @@ private boolean runShutdownHooks() {
         gracefulShutdownTimeout = unit.toNanos(timeout);
 
         if (oldState == ST_NOT_STARTED) {
-            doStartThread();
+            try {
+                doStartThread();
+            } catch (Throwable cause) {
+                STATE_UPDATER.set(this, ST_TERMINATED);
+                terminationFuture.tryFailure(cause);
+
+                if (!(cause instanceof Exception)) {
```

Comment:
optional:

`if (cause instanceof Error)) {`

Reply:
I think I would like to keep it as it is for now just in case someone creates a special `Throwable` sub-type